tests/text/test_tabstops.py

```diff
@@ -0,0 +1,36 @@
+# encoding: utf-8
+
+"""
+Test suite for the docx.text.parfmt module, containing the ParagraphFormat
+object.
+"""
+
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from docx.text.parfmt import ParagraphFormat
+
+import pytest
+
+from ..unitutil.cxml import element
+
+
+class DescribeTabStops(object):
+
+    def it_knows_its_length(self, len_fixture):
+        tab_stops, expected_value = len_fixture
+        assert len(tab_stops) == expected_value
+
+    # fixture --------------------------------------------------------
+
+    @pytest.fixture(params=[
+        ('w:p/w:pPr',                          0),
+        ('w:p/w:pPr/w:tabs/w:tab{w:pos=2880}', 1),
+    ])
+    def len_fixture(self, request):
+        tab_stops_cxml, expected_value = request.param
+        tab_stops = ParagraphFormat(element(tab_stops_cxml)).tab_stops
+        return tab_stops, expected_value
+
+    # fixture components ---------------------------------------------
```

Comment:
A couple things here. First, you should never generally have to involve the parent object (ParagraphFormat in this case) when unit testing one of its children. So while ParagraphFormat is used in tests/text/test_parfmt.py for accessing the tab_stops object, it shouldn't appear for doing anything *with* a TabStops object. Likely it wouldn't appear in this file at all.

So the XML starts with the element proxied by TabStops, which in this case is w:pPr.

Related to this, you should always be able to get your unit object by calling it's constructor. Having to use ParagraphFormat.tab_stops is a code smell because it relies on the proper functioning of that method (so the test isn't isolated).

See the merged version for more details.